posix/unix.c

```diff
@@ -373,7 +380,12 @@ int unix_accept4(unsigned socket, struct sockaddr *address, socklen_t *address_l
 			break;
 		}
 
-		if ((new = unixsock_alloc(&newid, s->type, nonblock)) == NULL) {
+		if (s->nonblock && (s->connect == NULL)) {
+			err = -EWOULDBLOCK;
+			break;
+		}
+
+		if ((new = unixsock_alloc(&newid, s->type, nonblock, 2)) == NULL) {
```

Comment:
Why are you setting ref counter to  2?

Reply:
See my previous comment - there was a race condition on socket creation. We add the socket to the tree (ref++) and also return reference to it to the creator (ref++). So initial ref counter should be set 2 references

Reply:
Perhaps a brief comment might clarify that for the future, though

Reply:
What race condition? Do you mean this line:
https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/572/files#diff-1c9efdfe25b04087e6b3c413cc19b4236acb47caf506607b871bffdb51344898R422

Reply:
> I think it would be best to create new unixsock with refcnt = 2 and require unixsock_put() at the end.
Rationale - we have a reference to the new object, that might be removed from the tree by someone else meanwhile. If we handle and use this reference, it should be reflected in the refcnt (1 for presence in the tree + 1 for handling it by us after creation).